main.go

```diff
@@ -35,6 +37,7 @@ func main() {
 	kingpin.Flag("debug", "Enable debug logging.").BoolVar(&config.Debug)
 	kingpin.Flag("pdb-name-suffix", "Specify default PDB name suffix.").Default(defaultPDBNameSuffix).StringVar(&config.PDBNameSuffix)
 	kingpin.Flag("non-ready-ttl", "Set the ttl for when to remove the managed PDB if the deployment/statefulset is unhealthy (default: disabled).").Default(defaultNonReadyTTL).DurationVar(&config.NonReadyTTL)
+	kingpin.Flag("use-parent-resource-hash", "Uses parent-resource-hash labels as selector for PDBs.").BoolVar(&config.ParentResourceHash)
```

Comment:
Should we maybe add logic to inject a custom selector into an annotation on deployments/statefulsets into the admission controller? This way we don't need to duplicate the hash logic, we can update it in just one place and it can be used by other users as well by simply setting it manually.

Reply:
I also thought about this, but in that case wouldn't it be easier to change the admission controller to simply inject a label in the pod template (of deployment and statefulset) and use this as opposed to having an extra annotation with a similar purpose? I understand it will do a rolling of the pods on the first time, but seems more simple in the long run.

The semantics of the pdb-controller flag would just be "use pod label `x`" as selector if present or fallback to normal selector. WDYT?